src/System.Management/tests/System/Management/ManagementClassTestsMofRequired.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Management.Tests
+{
+    [SkipOnTargetFramework(TargetFrameworkMonikers.Uap, "WMI not supported via UAP")]
+    [Collection("Mof Collection")]
+    public class ManagementClassTestsMofRequired
+    {
+        [ConditionalFact(typeof(WmiTestHelper), nameof(WmiTestHelper.IsElevatedAndNotNanoServer))]
+        public void Create_Modify_Delete_Static_Class()
+        {
+            using (var newClass = new ManagementClass(WmiTestHelper.Namespace))
+            {
+                const string NewClassName = "CoreFX_Create_Modify_Delete_Static_Class森鷗外";
+                const string PropertyName = "Key";
+                const int PropertyValue = 10;
+
+                newClass["__CLASS"] = NewClassName;
+                newClass.Properties.Add(PropertyName, CimType.SInt32, false);
+                newClass.Properties[PropertyName].Qualifiers.Add("key", true);
+                newClass.Put();
+
+                var targetClass = new ManagementClass(WmiTestHelper.Namespace, NewClassName, null);
+                targetClass.Get();
+
+                newClass[PropertyName] = PropertyValue;
+                newClass.Put();
+                targetClass.Get();
+                Assert.Equal(PropertyValue, (int)targetClass[PropertyName]);
+
+                // If any of the steps below fail it is likely that the new class was not deleted, likely it will have to
+                // be deleted via a tool like wbemtest.
+                newClass.Delete();
+                ManagementException managementException = Assert.Throws<ManagementException>(() => targetClass.Get());
+                Assert.Equal(ManagementStatus.NotFound, managementException.ErrorCode);
+            }
+        }
+
+        [ConditionalFact(typeof(WmiTestHelper), nameof(WmiTestHelper.IsElevatedAndNotNanoServer))]
+        public void Create_Modify_Delete_Static_And_Instance()
+        {
```

Comment:
these long tests are not really unit tests. but I get the impression that tests of WMI by its nature may not be beautiful. As long as they are reliable.

Reply:
These came from the few legacy tests that I found (like the process ones). These long ones they exercise features that need to be built upon: to set a property in a custom class one needs to create the class and the property.